### PR TITLE
added small improvement for release command

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

1. Added some more docs to the release increment calculation
2. I made it log what it does on DEBUG
3. I made sure we never return a lower version (i.e. adding the `.dev0`  tag is moving to a lower version

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [x ] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/inmanta/moduletool.py [hunk 1]
```diff
@@ -1107,24 +1112,63 @@ def _get_dev_version_with_minimal_distance_to_previous_stable_release(
...
+                    current_diff,
+                    minimal_version_bump_to_prev_release,
                 )
+                # It is good as it is
+                new_version = current_version
             else:
-                return new_version
-        else:
-            return VersionOperation.set_version_tag(current_version, version_tag="dev0")
+                LOGGER.debug(
```
Comment (wouterdb) [inline src/inmanta/moduletool.py:1150]: This is different

File: changelogs/unreleased/fix_release_command.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+description: Ensure the `inmanta module release` command never decreases the version number
+change-type: patch
+destination-branches: [master, iso6, iso5]
```
Comment (sanderr) [inline changelogs/unreleased/fix_release_command.yml:3]: I don't think the command exists on iso5
```suggestion
destination-branches: [master, iso6]
```

**Review by sanderr — APPROVED** [review]
Nice!

File: changelogs/unreleased/fix_release_command.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+description: Ensure the `inmanta module release` command never decreases the version number
```
Comment (sanderr) [inline changelogs/unreleased/fix_release_command.yml:1]: This sounds more ominous than it is, which might not be ideal if we use it as a changelog section?

File: changelogs/unreleased/fix_release_command.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+change-type: patch
+destination-branches: [master, iso6]
+sections:
+  minor-improvement:  "{{description}}"
```
Comment (wouterdb) [inline changelogs/unreleased/fix_release_command.yml:5]:
```suggestion
sections: []
```

File: changelogs/unreleased/fix_release_command.yml (hunk 3 above)
Comment (wouterdb) [reply changelogs/unreleased/fix_release_command.yml:1]: changed

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**